test/src/test/java/org/corfudb/runtime/CorfuRuntimeTest.java

```diff
@@ -173,4 +177,62 @@ public void doesNotAllowReadsAfterSealAndBeforeNewLayout() throws Exception {
                 .isInstanceOf(ExecutionException.class)
                 .hasRootCauseInstanceOf(WrongEpochException.class);
     }
+
+    /**
+     * Implement a SystemUnavailable networkExceptionHandler that shutdown the runtime after a certain timeout.
+     *
+     * The correct behaviour for this networkExceptionHandler is that, once the router is disconnected during the whole timeout,
+     * a SystemUnavailableException is thrown and the runtime is shutdown.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void customNetworkExceptionHandler() throws Exception {
+        class TimeoutHandler {
+            CorfuRuntime rt;
+            long maxTimeout;
+            ThreadLocal<Long> localTimeStart = new ThreadLocal<>();
```

Comment:
Well this is an example, so I guess there is some freedom on how to implement it. I also think different threads could be interacting with different routers. If one thread is only reading, its queries will only go to the chain tail, and it will reset the timer each time. This could mask a problem with writes that could never go in.